sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Tests/E2EAzureMonitorDistroTests.cs

```diff
@@ -77,6 +80,44 @@ public async Task ValidateTelemetryExport()
             // TODO: This test needs to assert telemetry content. (ie: sample rate)
         }
 
+        [Theory]
+        [InlineData(null)]
+        [InlineData("true")]
+        [InlineData("false")]
+        public void ValidateLogFilteringProcessorIsAddedToLoggerProvider(string enableLogSampling)
+        {
+            try
+            {
+                Environment.SetEnvironmentVariable("OTEL_DOTNET_AZURE_MONITOR_EXPERIMENTAL_ENABLE_LOG_SAMPLING", enableLogSampling);
+
+                var sv = new ServiceCollection();
+                sv.AddOpenTelemetry().UseAzureMonitor(o => o.ConnectionString = "InstrumentationKey=00000000-0000-0000-0000-000000000000");
+
+                var sp = sv.BuildServiceProvider();
+                var loggerProvider = sp.GetRequiredService<ILoggerProvider>();
+                var sdkProvider = typeof(OpenTelemetryLoggerProvider).GetField("Provider", BindingFlags.Instance | BindingFlags.NonPublic)?.GetValue(loggerProvider);
+                var processor =  sdkProvider?.GetType().GetProperty("Processor", BindingFlags.Instance | BindingFlags.Public)?.GetMethod?.Invoke(sdkProvider, null);
+
+                if (processor != null)
+                {
+                    if (enableLogSampling == "true")
+                    {
+                        Assert.True(processor is LogFilteringProcessor);
+                        Assert.True(processor is BatchLogRecordExportProcessor);
+                    }
+                    else
+                    {
+                        Assert.True(processor is not LogFilteringProcessor);
+                        Assert.True(processor is BatchLogRecordExportProcessor);
+                    }
+                }
```

Comment:
I'm thinking we need more cases. 
`else if processor == null`

Reply:
added assert